compiler/ballerina-backend-llvm/src/main/ballerina/src/compiler_backend_llvm/llvm_bb_body_gen.bal

```diff
@@ -37,19 +37,33 @@ type BbBodyGenrator object {
             self.genBinaryOpIns(instruction);
         } else if (instruction is bir:ConstantLoad) {
             self.genConstantLoadIns(instruction);
+        } else if (instruction is bir:TypeCast) {
+            self.genTypeCast(instruction);
+        } else if (instruction is bir:TypeTest) {
+            self.genTypeTest(instruction);
         }
     }
 
-    function genMoveIns(bir:Move moveIns) {
-        llvm:LLVMValueRef lhsRef = self.parent.getLocalVarRefById(moveIns.lhsOp.variableDcl.name.value);
-        var rhsVarOp = moveIns.rhsOp;
-        llvm:LLVMValueRef rhsVarOpRef = self.parent.genLoadLocalToTempVar(rhsVarOp);
-        var loaded = <llvm:LLVMValueRef> llvm:llvmBuildStore(self.builder, rhsVarOpRef, lhsRef);
+    function genTypeTest(bir:TypeTest typeTestIns) {
+        bir:BType rhsOpType = typeTestIns.rhsOp.typeValue;
+        boolean isLhsUnionType = isUnionType(rhsOpType);
+        if (isLhsUnionType) {
+            llvm:LLVMValueRef unionTagValue =
+                        self.parent.loadElementFromStruct(typeTestIns.rhsOp, TAGGED_UNION_FLAG_INDEX);
+            llvm:LLVMValueRef valueOfType = getValueRefFromInt(getTag(typeTestIns.typeValue), 0);
+            llvm:LLVMValueRef isEq = llvm:llvmBuildICmp(self.builder, llvm:LLVMIntEQ, unionTagValue, valueOfType, "typeTest");
+            _ = llvm:llvmBuildStore(self.builder, isEq, self.parent.getLocalVarRef(typeTestIns.lhsOp));
```

Comment:
Looks like the return value of the `llvmBuildStore(...)` method is never used at any place. So Im wondering whether this API actually needs a return?

Reply:
This api binds bytedecos LLVMBuildStore function. It has a return value hence we have a return value for the extern function as well. WDYT the best way?

Reply:
since only the compiler is using this we can decide on the ballerina API, and it doesn't have to be exactly same as the native API. If we know for sure that the return value is/will **never** used, then its clean to have a API without returns.